statsmodels/sandbox/mice/mice.py

```diff
@@ -0,0 +1,1167 @@
+    mean structures containing main effects of all other variables in
+    `data`.  The variable named `x1` has a condtional mean structure
+    that includes an additional term for x2^2.
+    %(_mice_data_example_1)s
+    Notes
+    -----
+    Allowed perturbation methods are 'gaussian' (the model parameters
+    are set to a draw from the Gaussian approximation to the posterior
+    distribution), and 'boot' (the model parameters are set to the
+    estimated values obtained when fitting a bootstrapped version of
+    the data set).
+
+    `history_func` can be implemented to have side effects such as
+    saving the current imputed data set to disk.
+    """ % {'_mice_data_example_1': _mice_data_example_1}
+
+    def __init__(self, data, perturbation_method='gaussian',
+                 history_func=None):
+
+        # Drop observations where all variables are missing.  This
+        # also has the effect of copying the data frame.
+        self.data = data.dropna(how='all').reset_index(drop=True)
+
+        self.history_func = history_func
+        self.history = []
+
+        self.perturbation_method = defaultdict(lambda :
+                                               perturbation_method)
+
+        # Map from variable name to indices of observed/missing
+        # values.
+        self.ix_obs = {}
+        self.ix_miss = {}
+        for col in self.data.columns:
+            ix_obs, ix_miss = self._split_indices(self.data[col])
+            self.ix_obs[col] = ix_obs
+            self.ix_miss[col] = ix_miss
+
+        # Most recent model and results for each variable.
+        self.models = {}
+        self.results = {}
+
+        # Map from variable names to the conditional formula.
+        self.conditional_formula = {}
+
+        # Map from variable names to init/fit args of the conditional
+        # models.
+        self.init_args = defaultdict(lambda : {})
+        self.fit_args = defaultdict(lambda : {})
+
+        # Map from variable name to the method used to handle the
+        # scale parameter of the conditional model.
+        self.scale_method = defaultdict(lambda : "fix")
+
+        # Map from variable names to the model class.
+        self.model_class = {}
+
+        # Map from variable names to most recent params update.
+        self.params = {}
+
+        # Set default imputers.
+        for vname in data.columns:
+            self.set_imputer(vname)
+
+        # The order in which variables are imputed in each cycle.
+        # Impute variables with the fewest missing values first.
+        vnames = data.columns.tolist()
+        nmiss = [len(self.ix_miss[v]) for v in vnames]
+        nmiss = np.asarray(nmiss)
+        ii = np.argsort(nmiss)
+        ii = ii[sum(nmiss == 0):]
+        self.cycle_order = [vnames[i] for i in ii]
+
+        # Fill missing values with column-wise mean.  This is the
+        # starting imputation.
+        self.data = self.data.fillna(self.data.mean())
+
+        self.k_pmm = 20
```

Comment:
pure initialization might be  better with `None`, let the method handle the default?